network/test/unicast_authorization_test.go

```diff
@@ -408,3 +394,57 @@ func (u *UnicastAuthorizationTestSuite) TestUnicastAuthorization_PublicChannel()
 	// wait for slashing violations consumer mock to invoke run func and close ch if expected method call happens
 	unittest.RequireCloseBefore(u.T(), u.waitCh, u.channelCloseDuration, "could close ch on time")
 }
+
+// TestUnicastAuthorization_UnauthorizedUnicastOnChannel tests that messages sent via unicast that are not authorized for unicast are rejected.
+func (u *UnicastAuthorizationTestSuite) TestUnicastAuthorization_UnauthorizedUnicastOnChannel() {
+	// setup mock slashing violations consumer and middlewares
+	slashingViolationsConsumer := mocknetwork.NewViolationsConsumer(u.T())
+	u.setupMiddlewaresAndProviders(slashingViolationsConsumer)
+
+	// set sender id role to RoleConsensus to avoid unauthorized sender validation error
+	u.senderID.Role = flow.RoleConsensus
+
+	expectedSenderPeerID, err := unittest.PeerIDFromFlowID(u.senderID)
+	require.NoError(u.T(), err)
+
+	expectedViolation := &slashing.Violation{
+		Identity:  u.senderID,
+		PeerID:    expectedSenderPeerID.String(),
+		MsgType:   "BlockProposal",
+		Channel:   channels.ConsensusCommittee,
+		IsUnicast: true,
+		Err:       message.ErrUnauthorizedUnicastOnChannel,
+	}
+
+	slashingViolationsConsumer.On(
+		"OnUnauthorizedUnicastOnChannel",
+		expectedViolation,
+	).Once().Run(func(args mockery.Arguments) {
+		close(u.waitCh)
+	})
+
+	overlay := mocknetwork.NewOverlay(u.T())
+	overlay.On("Identities").Maybe().Return(func() flow.IdentityList {
+		return u.providers[0].Identities(filter.Any)
+	})
+	overlay.On("Topology").Maybe().Return(func() flow.IdentityList {
+		return u.providers[0].Identities(filter.Any)
+	}, nil)
+	overlay.On("Identity", mock.AnythingOfType("peer.ID")).Return(u.senderID, true)
```

Comment:
can you make this more precise?
```suggestion
	overlay.On("Identity", expectedSenderPeerID).Return(u.senderID, true)
```

Reply:
https://github.com/onflow/flow-go/pull/3378/commits/6e31ccefc05c9984598311ec73f9bf28bffef4de